In [ ]:
library(tidyverse) # install this package if needed
library(repr)

In [12]:
players <- read_csv('data/players.csv')
sessions <- read_csv('data/sessions.csv')

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [26]:
glimpse(players)
glimpse(sessions)
dim(players)
dim(sessions)
colSums(is.na(players))
colSums(is.na(sessions))
summary(players)
summary(sessions)


Rows: 196
Columns: 7
$ experience   <chr> "Pro", "Veteran", "Veteran", "Amateur", "Regular", "Amate…
$ subscribe    <lgl> TRUE, TRUE, FALSE, TRUE, TRUE, TRUE, TRUE, FALSE, TRUE, T…
$ hashedEmail  <chr> "f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8…
$ played_hours <dbl> 30.3, 3.8, 0.0, 0.7, 0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 1.6, 0…
$ name         <chr> "Morgan", "Christian", "Blake", "Flora", "Kylie", "Adrian…
$ gender       <chr> "Male", "Male", "Male", "Female", "Male", "Female", "Fema…
$ Age          <dbl> 9, 17, 17, 21, 21, 17, 19, 21, 47, 22, 23, 17, 25, 22, 17…
Rows: 1,535
Columns: 5
$ hashedEmail         <chr> "bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8a…
$ start_time          <chr> "30/06/2024 18:12", "17/06/2024 23:33", "25/07/202…
$ end_time            <chr> "30/06/2024 18:24", "17/06/2024 23:46", "25/07/202…
$ original_start_time <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+12…
$ original_end_time   <dbl> 1.71977e+12, 1.71867e+12, 1.72193e+12, 1.72188e+

[1] 196   7

[1] 1535    5

experience    subscribe  hashedEmail played_hours         name       gender 
           0            0            0            0            0            0 
         Age 
           2

hashedEmail          start_time            end_time original_start_time 
                  0                   0                   2                   0 
  original_end_time 
                  2

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

**<h2>Data Overview</h2>**

<p>This project uses the two datasets <code>players.csv</code> and <code>sessions.csv</code> which contain information about individual player's and their play sessions.</p> 

**<code>players.csv</code>**
- **Number of observations:** 196 
- **Number of variables:** 7
- **Description:** Contains general background information on the player
- **Purpose:** Provides statistics like experience, age, gender, hours played that may help explain engagement or performance patterns.

| Variable Name  | Type      | Description                                                 |
| -------------- | --------- | ----------------------------------------------------------- |
| `hashedEmail`  | Character | Unique identifier for each player                           |
| `name`         | Character | Player name                                                 |
| `gender`       | Character | Self-reported gender (7 levels)                             |
| `Age`          | Double    | Age in years                                                |
| `experience`   | Character | Player experience level (e.g., Amateur, Intermediate, etc.) |
| `subscribe`    | Boolean   | Whether the player subscribes to a game-related newsletter  |
| `played_hours` | Double    | Total hours played                                          |




**<code>sessions.csv</code>**
- **Number of observations:** 1,535
- **Number of variables:** 5
- **Description:** Each row represents a single gameplay session
- **Purpose:** Tracks when sessions occured 

| Variable Name         | Type      | Description                           |
| --------------------- | --------- | ------------------------------------- |
| `hashedEmail`         | Character | Unique ID linking sessions to players |
| `start_time`          | Character | Session start time                    |
| `end_time`            | Character | Session end time                      |
| `original_start_time` | Double    | Unix timestamp for start              |
| `original_end_time`   | Double    | Unix timestamp for end                |




**<h5>Summary Statistics</h5>**

**<code>players.csv</code>**

**Numeric:**

| Variable       | Mean  | SD    | Min  | Median | Max    |
| -------------- | ----- | ----- | ---- | ------ | ------ |
| `Age`          | 21.14 | 7.39  | 9.00 | 19.00  | 58.00  |
| `played_hours` | 5.85  | 28.36 | 0.00 | 0.10   | 223.10 |

**Non-numeric:**

| Variable              | Type        | Description             | Notes                                      |
| --------------------- | ----------- | ----------------------- | ------------------------------------------ |
| `gender`              | Categorical | Self-reported gender    | 7 unique values; majority “Male” (124/196) |
| `experience`          | Categorical | Player experience level | 5 levels; most frequent “Amateur”          |
| `subscribe`           | Boolean     | Subscription status     | 144 TRUE, 52 FALSE                         |
| `hashedEmail`, `name` | Character   | Identifiers             | All unique; not useful for modeling        |

**<code>sessions.csv</code>**

**Numeric:**

| Variable              | Mean       | SD       | Min        | Median     | Max        |
| --------------------- | ---------- | -------- | ---------- | ---------- | ---------- |
| `original_start_time` | 1.719×10¹² | 3.56×10⁹ | 1.712×10¹² | 1.719×10¹² | 1.727×10¹² |
| `original_end_time`   | 1.719×10¹² | 3.55×10⁹ | 1.712×10¹² | 1.719×10¹² | 1.727×10¹² |

**Non-numeric:**

| Variable                 | Type                 | Description           | Notes                                      |
| ------------------------ | -------------------- | --------------------- | ------------------------------------------ |
| `hashedEmail`            | Character            | Player ID key         | 125 unique values; used to join to players |
| `start_time`, `end_time` | Character (datetime) | Session start and end | 2 missing end times                        |


**Potential Issues in Data**
- in ```players.csv```: ```Age``` has 2 missing values
- in ```sessions.csv```: end_time has 2 missing values
- in ```players.csv```: ```played_hours``` is skewed with most values near 0
- in ```sessions.csv```: ```start_time``` and ```end_time``` are stored as characters, not a date object
- in ```players.csv``` and ```sessions.csv```: looking at ```hashedEmail``` (present in both datasets), we can see that only 125/196 players have session records.
- not all players have recorded sessions (sampling bias) 
- self-reported ages and genders may not be accurate
- timestamps may not be accurate (inconsistent timezones, system clock errors)



**Data Collection**
- Data appears to be collected via an automated software which tracks player data and session data.
- ```players.csv``` most likely comes from user registration / profile. 
- According to the research group's website, gameplay, speech, and key presses are all recorded.
- The timeframe appears to be from June-July 2024.
- Emails have been hidden via hashing.